Source/WebCore/loader/ImageLoader.cpp

```diff
@@ -206,106 +211,160 @@ void ImageLoader::updateFromElement(RelevantMutation relevantMutation)
     // Do not load any image if the 'src' attribute is missing or if it is
     // an empty string.
     CachedResourceHandle<CachedImage> newImage;
-    if (!attr.isNull() && !StringView(attr).containsOnly<isASCIIWhitespace<UChar>>()) {
-        ResourceLoaderOptions options = CachedResourceLoader::defaultCachedResourceOptions();
-        options.contentSecurityPolicyImposition = protectedElement()->isInUserAgentShadowTree() ? ContentSecurityPolicyImposition::SkipPolicyCheck : ContentSecurityPolicyImposition::DoPolicyCheck;
-        options.loadedFromPluginElement = is<HTMLPlugInElement>(element()) ? LoadedFromPluginElement::Yes : LoadedFromPluginElement::No;
-        options.sameOriginDataURLFlag = SameOriginDataURLFlag::Set;
-        options.serviceWorkersMode = is<HTMLPlugInElement>(element()) ? ServiceWorkersMode::None : ServiceWorkersMode::All;
-        RefPtr imageElement = dynamicDowncast<HTMLImageElement>(element());
+    if (attr.isNull() || StringView(attr).containsOnly<isASCIIWhitespace<UChar>>()) {
+        if (!attr.isNull()) {
+            m_failedLoadURL = attr;
+            m_hasPendingErrorEvent = true;
+            loadEventSender().dispatchEventSoon(*this, eventNames().errorEvent);
+        }
+        didUpdateCachedImage(relevantMutation, WTFMove(newImage));
+        return;
+    }
```

Comment:
I think making this early return is a clean-up step. Can we start by making it in a separate PR? Also please let make things clear even if they are verbose a little:
```
if (StringView(attr).containsOnly<isASCIIWhitespace<UChar>>()) {
    ...
    didUpdateCachedImage(relevantMutation, WTFMove(newImage));
    return;
}

if (!attr.isNull()) {
    didUpdateCachedImage(relevantMutation, WTFMove(newImage));
    return;
}
```